### PR TITLE
Icon: Auto-generate readme

### PR DESCRIPTION
**Pull request by mirka**
Companion to #67280

## What?

Switch the README for `Icon` to an auto-generated one (see also #66035).

## Why?

To improve accuracy and maintenance cost.

## Testing Instructions

`npm run docs:components` to verify the changes to the README.

### CODE REVIEW

File: packages/components/src/icon/README.md [hunk 1]
```diff
@@ -1,82 +1,39 @@
...
+<p class="callout callout-info">See the <a href="https://wordpress.github.io/gutenberg/?path=/docs/components-icon--docs">WordPress Storybook</a> for more detailed, interactive documentation.</p>
 
-#### With a Dashicon
+Renders a raw icon without any initial styling or wrappers.
 
 ```jsx
-import { Icon } from '@wordpress/components';
+import { wordpress } from '@wordpress/icons';
 
-const MyIcon = () => <Icon icon="screenoptions" />;
```
Comment (mirka) [inline packages/components/src/icon/README.md:12]: These usage examples are moved to the Storybook in #67280.

Instead, the most common use case (using an icon from `@wordpress/icons`) is highlighted in the main code snippet.

File: packages/components/src/icon/index.tsx [hunk 2]
```diff
@@ -25,30 +25,45 @@ export type IconType =
 	| ( ( props: { size?: number } ) => JSX.Element )
 	| JSX.Element;
 
-interface BaseProps {
```
Comment (mirka) [inline packages/components/src/icon/index.tsx:28]: Removed this unnecessary layer of indirection, and inlined the prop types.

**Review by tyxla — APPROVED** [review]
Autogenerated READM looks great 👍 🚀 

Some stories seem broken, but I'll review #67280 separately, maybe you've addressed them there already.

File: packages/components/src/icon/README.md [hunk 3]
```diff
@@ -1,82 +1,39 @@
...
-);
-```
-
-#### Specifying a className
-
-```jsx
-import { Icon } from '@wordpress/components';
-
-const MyIcon = () => <Icon icon="screenoptions" className="example-class" />;
-```
```
Comment (tyxla) [inline packages/components/src/icon/README.md:62]: I don't see any mention of `className` in #67280, nor here. Is that intentional?

Comment (mirka) [reply packages/components/src/icon/README.md:62]: Good question. We currently have kind of a mix, where some components pass through all rest props and some only pass through specific props (like `className`). In the former case, it is unreasonable to list out all possible props as allowed in TypeScript (`id`, `aria-*`, etc) and `react-docgen-typescript` suppresses them too by default, unless it's required or we add an explicit prop description.

I think almost all of our components pass through at least `className`, so it should be one of those things that just work and don't need to be mentioned for every component.

**Pull request merged**